### DISCUSSION TITLE
How to use old versions of php

### DISCUSSION THREAD

**Discussion by PierW** [laravel/valet#1396] [General]
If I tried to use php@5.6
`valet use php@5.6`
I have this error:
**_Valet doesn't support PHP version: php@5.6 (try something like 'php@8.1' instead)_**
I have to work on a project with php 5.6, how can I do?

**Comment by ihorvorotnov** [comment]
Here's a list of supported PHP versions:
https://github.com/laravel/valet/blob/f65774d0705a82b0c0aa670152ab2d24f2f347ea/cli/Valet/Brew.php#L11-L20

**Reply by PierW** [reply]
thanks @ihorvorotnov! So, how do you suggest to work with old projects that requires php 5.6 if we just use laravel valet locally?

For Example, with **Laravel Valet 4** we have problems (warnings) with php 7.1 - 7.2 when we have to run `composer global update` and switch from php 8 (composer 2) to php 7 (composer 1):
laravel/valet v4.0.1 requires illuminate/collections ^8.0|^9.0|^10.0

So the solution for me is to add the flag: _--ignore-platform-reqs_
so:
`composer global update --ignore-platform-reqs`

I tried to write more info that I can in this discussions to help also other persons.

Maybe a solution to use php 5.6 locally (avoiding breaking valet configurations) is to install manually via brew php 5.6 and configure httpd and not nginx? so will it be enough to disable nginx and enable httpd??

**Comment by ihorvorotnov** [comment]
IIRC older versions of Valet should support PHP 5.6 installed via Homebrew from `shivammathur/php`. You can just set `"laravel/valet": "^3.3"` in your `~/.composer/composer.json` file and run `composer update` followed by `valet install`.

For reference, here's the commit that removed 5.6 support: eefc06b
